src/Rules/Comparison/ImpossibleCheckTypeFunctionCallRule.php

```diff
@@ -41,17 +42,21 @@ public function processNode(Node $node, Scope $scope): array
 		if (strtolower($functionName) === 'is_a') {
 			return [];
 		}
-		$isAlways = $this->impossibleCheckTypeHelper->findSpecifiedType($scope, $node);
+		[$isAlways, $reasons] = $this->impossibleCheckTypeHelper->findSpecifiedType($scope, $node);
 		if ($isAlways === null) {
 			return [];
 		}
 
-		$addTip = function (RuleErrorBuilder $ruleErrorBuilder) use ($scope, $node): RuleErrorBuilder {
+		$addTip = function (RuleErrorBuilder $ruleErrorBuilder) use ($scope, $node, $reasons): RuleErrorBuilder {
+			if (count($reasons) > 0) {
+				return $ruleErrorBuilder->acceptsReasonsTip($reasons);
```

Comment:
I copied https://github.com/phpstan/phpstan-src/compare/6ac62d3b00f64e72c02b7aa477e425e90f46fa65...dc5d8f4d3eef18b1d80b8ffb3a1adfe8de6d7268#diff-6f5d047ab5b60f37c6dca68882f80e1f7ba2d946e0ffb91290d39589cf55acc2R58-R60

but looking at the failing test, I feel like we want to keep the `treatPhpDocTypesAsCertainTip` @ondrejmirtes , no ?
(So maybe the StrictComparisonOfDifferentTypesRule too ?)

Currently I dunno if I need to
- Update the tests.
- Or update this line of code.

Your call.